sdk/spring-sdk/src/test/scala/kalix/springsdk/impl/SpringSdkMessageCodecSpec.scala

```diff
@@ -0,0 +1,196 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.springsdk.impl
+
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import com.fasterxml.jackson.annotation.JsonTypeName
+import kalix.springsdk.impl.SpringSdkMessageCodecSpec.AnnotatedWithSingleName.Elephant
+import kalix.springsdk.impl.SpringSdkMessageCodecSpec.SimpleClass
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+object SpringSdkMessageCodecSpec {
+
+  @JsonCreator
+  case class SimpleClass(str: String, in: Int)
+
+  object AnnotatedWithSingleName {
+
+    @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+    @JsonSubTypes(
+      Array(
+        new JsonSubTypes.Type(value = classOf[Lion], name = "lion"),
+        new JsonSubTypes.Type(value = classOf[Elephant], name = "elephant")))
+    sealed trait Animal
+
+    final case class Lion(name: String) extends Animal
+
+    final case class Elephant(name: String, age: Int) extends Animal
+
+  }
+
+  object AnnotatedWithManyNames {
+
+    @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+    @JsonSubTypes(
+      Array(
+        new JsonSubTypes.Type(value = classOf[Lion], names = Array("lion", "noil")),
+        new JsonSubTypes.Type(value = classOf[Elephant], names = Array("elephant", "tnahpele"))))
+    sealed trait Animal
+
+    final case class Lion(name: String) extends Animal
+
+    final case class Elephant(name: String, age: Int) extends Animal
+
+  }
+  object AnnotatedWIthJsonTypeName {
+
+    @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+    @JsonSubTypes(Array(new JsonSubTypes.Type(value = classOf[Lion]), new JsonSubTypes.Type(value = classOf[Elephant])))
+    sealed trait Animal
+
+    @JsonTypeName("lion")
+    final case class Lion(name: String) extends Animal
+    @JsonTypeName("elephant")
+    final case class Elephant(name: String, age: Int) extends Animal
+  }
+
+  object HierarchyWithAbstractClass {
+
+    @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+    @JsonSubTypes(
+      Array(
+        new JsonSubTypes.Type(value = classOf[Lion], name = "lion"),
+        new JsonSubTypes.Type(value = classOf[Elephant], name = "elephant")))
+    abstract class Animal
+
+    final case class Lion(name: String) extends Animal
+
+    final case class Elephant(name: String, age: Int) extends Animal
+
+  }
+}
+class SpringSdkMessageCodecSpec extends AnyWordSpec with Matchers {
+
+  "The SpringSdkMessageCodec" should {
+
+    "by default FQCN for typeUrl (java)" in {
+      val encoded = SpringSdkMessageCodec.encodeJava(SimpleClass("abc", 10))
+      encoded.getTypeUrl shouldBe "json.kalix.io/kalix.springsdk.impl.SpringSdkMessageCodecSpec$SimpleClass"
```

Comment:
maybe using here the variable `kalix.javasdk.KALIX_JSON` to show where this comes from?